File: app/src/main/java/com/example/miniapp/helper_classes/NotificationHelper.java
package com.example.miniapp.helper_classes;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.ContextWrapper;
import android.content.Intent;
import android.graphics.Color;
import android.util.Log;

import androidx.core.app.NotificationCompat;

import com.example.miniapp.R;
import com.example.miniapp.views.MainActivity;

public class NotificationHelper extends ContextWrapper {
    private static final String PRIMARY_CHANNEL_ID = "primary_notification_channel";
    private NotificationManager notificationManager;

    public NotificationHelper(Context base) {
        super(base);
        createNotificationChannel();
    }

    private void createNotificationChannel(){
        notificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            NotificationChannel notificationChannel = new NotificationChannel(PRIMARY_CHANNEL_ID, "Mascot Notification", NotificationManager.IMPORTANCE_HIGH);
            notificationChannel.enableLights(true);
            notificationChannel.setLightColor(Color.YELLOW);
            notificationChannel.enableVibration(true);
            notificationChannel.setDescription("Notification from Mascot");
            notificationManager.createNotificationChannel(notificationChannel);
        }
    }

    private NotificationCompat.Builder getNotificationBuilder(String task){
        Intent openAppIntent = new Intent(getApplicationContext(), MainActivity.class);
        PendingIntent pendingIntent = PendingIntent.getActivity(getApplicationContext(), 0, openAppIntent, PendingIntent.FLAG_ONE_SHOT);

        NotificationCompat.Builder notifyBuilder = new NotificationCompat.Builder(this, PRIMARY_CHANNEL_ID)
                .setContentTitle("To-do:")
                .setContentText(task)
                .setContentIntent(pendingIntent)
                .setSmallIcon(R.drawable.ic_alarm_clock);

        return notifyBuilder;
    }


    // TODO: add pending intent to do something when notification is clicked
    public void sendNotification(int notificationID, String task) {
        if (notificationID == 0){
            Log.v("MY TAG", "Error in sending notification");
        }
        NotificationCompat.Builder notifyBuilder = getNotificationBuilder(task);
        notificationManager.notify(notificationID, notifyBuilder.build());
    }
}

File: app/src/main/java/com/example/miniapp/helper_classes/NotificationHelper.kt
package com.example.miniapp.helper_classes

import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.ContextWrapper
import android.content.Intent
import android.graphics.Color
import android.os.Build
import androidx.core.app.NotificationCompat
import com.example.miniapp.R
import com.example.miniapp.views.MainActivity

class NotificationHelper(base: Context?) : ContextWrapper(base) {
    private var notificationManager: NotificationManager? = null

    companion object {
        private const val PRIMARY_CHANNEL_ID = "primary_notification_channel"
    }

    init {
        createNotificationChannel()
    }

    private fun createNotificationChannel() {
        notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val notificationChannel = NotificationChannel(PRIMARY_CHANNEL_ID, "Mascot Notification", NotificationManager.IMPORTANCE_HIGH)
            notificationChannel.enableLights(true)
            notificationChannel.lightColor = Color.YELLOW
            notificationChannel.enableVibration(true)
            notificationChannel.description = "Notification from Mascot"
            notificationManager!!.createNotificationChannel(notificationChannel)
        }
    }

    private fun getNotificationBuilder(task: String?): NotificationCompat.Builder {
        val openAppIntent = Intent(applicationContext, MainActivity::class.java)
        val pendingIntent = PendingIntent.getActivity(applicationContext, 0, openAppIntent, PendingIntent.FLAG_UPDATE_CURRENT)
        return NotificationCompat.Builder(this, PRIMARY_CHANNEL_ID)
                .setContentTitle("To-do:")
                .setContentText(task)
                .setContentIntent(pendingIntent)
                .setSmallIcon(R.drawable.ic_alarm_clock)
    }

    fun sendNotification(notificationID: Int, task: String?) {
        if (notificationID == 0) {
            log("Error in sending notification")
        }
        val notifyBuilder = getNotificationBuilder(task)
        notificationManager!!.notify(notificationID, notifyBuilder.build())
    }
}